tests/docker/basics/basics_test.go

```diff
@@ -0,0 +1,121 @@
+			}, "60s", "5s").Should(Succeed())
+			Eventually(func() error {
+				return tester.NodesReady(config.KubeconfigFile)
+			}, "40s", "5s").Should(Succeed())
+		})
+	})
+
+	Context("Verify Binaries and Images", func() {
+		It("has valid bundled binaries", func() {
+			for _, server := range config.Servers {
+				Expect(tester.VerifyValidVersion(server.Name, "kubectl")).To(Succeed())
+				Expect(tester.VerifyValidVersion(server.Name, "ctr")).To(Succeed())
+				Expect(tester.VerifyValidVersion(server.Name, "crictl")).To(Succeed())
+			}
+		})
+		It("has valid airgap images", func() {
+			Expect(config).To(Not(BeNil()))
+			err := VerifyAirgapImages(config.Servers)
+			Expect(err).NotTo(HaveOccurred())
+		})
+	})
+
+	Context("Use Local Storage Volume", func() {
+		It("should apply local storage volume", func() {
+			const volumeTestManifest = "../resources/volume-test.yaml"
+
+			// Apply the manifest
+			cmd := fmt.Sprintf("kubectl apply -f %s --kubeconfig=%s", volumeTestManifest, config.KubeconfigFile)
+			_, err := tester.RunCommand(cmd)
+			Expect(err).NotTo(HaveOccurred(), "failed to apply volume test manifest")
+		})
+		It("should validate local storage volume", func() {
+			Eventually(func() (bool, error) {
+				return tester.PodReady("volume-test", "kube-system", config.KubeconfigFile)
+			}, "20s", "5s").Should(BeTrue())
+		})
+	})
+})
+
+var failed bool
+var _ = AfterEach(func() {
+	failed = failed || CurrentSpecReport().Failed()
+})
+
+var _ = AfterSuite(func() {
+	if config != nil && !failed {
+		config.Cleanup()
+	}
+})
+
+// VerifyAirgapImages checks for changes in the airgap image list
+func VerifyAirgapImages(servers []tester.ServerConfig) error {
+	// This file is generated during the build packaging step
+	const airgapImageList = "../../../scripts/airgap/image-list.txt"
+
+	// Use a map to automatically handle duplicates
+	imageSet := make(map[string]struct{})
+	// Collect all images from containers
+	for _, server := range servers {
+		name := server.Name
+		cmd := fmt.Sprintf("docker exec %s crictl images -o json | jq -r '.images[].repoTags[0] | select(. != null)'", name)
+		output, err := tester.RunCommand(cmd)
+		Expect(err).NotTo(HaveOccurred(), "failed to execute crictl and jq: %v", err)
+
+		for _, line := range strings.Split(strings.TrimSpace(string(output)), "\n") {
+			if line != "" {
+				imageSet[line] = struct{}{}
+			}
+		}
+	}
+
+	// Convert map keys to slice
+	uniqueImages := make([]string, 0, len(imageSet))
+	for image := range imageSet {
+		uniqueImages = append(uniqueImages, image)
+	}
+
+	existing, err := os.ReadFile(airgapImageList)
+	if err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to read airgap list file: %v", err)
+	}
+
+	// Sorting doesn't matter with ContainElements
+	existingImages := strings.Split(strings.TrimSpace(string(existing)), "\n")
+	Expect(existingImages).To(ContainElements(uniqueImages))
```

Comment:
I think we want `HaveExactElements` here, to ensure that it also doesn't contain any NEW images that are missing from the list. That's what the current diff provides.